lib/api/core/plugins/pluginsManager.js

```diff
@@ -150,59 +195,49 @@ class PluginsManager {
           initTimeout
         } = this.config.common;
 
-      if (plugin.config.threads) {
-        debug('[%s] starting worker with %d threads', pluginName, plugin.config.threads);
-
-        return initWorkers(plugin, pluginName, this.kuzzle.config)
-          .timeout(initTimeout)
-          .catch(err => {
-            // eslint-disable-next-line no-console
-            this.silent || console.log('Plugin', pluginName, ' errored with message: ', err.message, '. Skipping...');
-            return Bluebird.resolve();
-          });
+      if (Object.keys(pluginsWorker).indexOf(plugin.name) > -1) {
+        // Do not load worker plugins as regular ones
+        return Bluebird.resolve();
       }
 
-      debug('[%s] starting plugin in "%s" mode', pluginName, plugin.config.privileged ? 'privileged' : 'standard');
+      debug('[%s] starting plugin in "%s" mode', plugin.name, plugin.config.privileged ? 'privileged' : 'standard');
 
       // We sandbox the Promise.resolve into a Promise to be able to catch throwed errors if any
-      return new Bluebird(resolve => {
+      return new Bluebird((resolve, reject) => {
         // Promise.resolve allows to convert any type of return into a bluebird Promise.
         // It allows to apply a timeout on it
-        return Bluebird.resolve(plugin.object.init(
+        Bluebird.resolve(plugin.object.init(
           plugin.config,
-          plugin.config.privileged ? new PrivilegedPluginContext(this.kuzzle, pluginName) : new PluginContext(this.kuzzle, pluginName)
+          plugin.config.privileged ? new PrivilegedPluginContext(this.kuzzle, plugin.name) : new PluginContext(this.kuzzle, plugin.name)
         ))
           .timeout(initTimeout)
           .then(initStatus => {
             if (initStatus === false) {
-              return Bluebird.reject(new Error(`Something went wrong during the initialization of the plugin ${pluginName}.`));
+              return reject(new PluginImplementationError(`Something went wrong during initialization of "${plugin.name}" plugin.`));
             }
-            if (plugin.object.hooks) {
-              initHooks(plugin, this.kuzzle);
+
+            if (plugin.object.controllers && !initControllers(this, plugin)) {
```

Comment:
should be simplified by:
```javascript
if (plugin.object.controllers) {
  initControllers(this, plugin);
}
```

And let `initControllers` method to throw an explicit `PluginImplementationError`, like it is done for `initHooks` or `initPipes` below